imagedephi/redact/redact.py

```diff
@@ -35,14 +35,17 @@ def get_base_rules():
 
 
 def iter_image_files(directory: Path) -> Generator[Path, None, None]:
-    """
-    Given a directory return an iterable of available images.
-
-    May raise a PermissionError if the directory is not readable.
-    """
+    """Given a directory return an iterable of available images."""
     for child in directory.iterdir():
-        if child.suffix in FILE_EXTENSION_MAP:
-            yield child
+        # Use first four bits to check if its a tiff file
+        if child.is_file():
+            try:
+                data = open(child, "rb").read(4)
+            except PermissionError:
+                pass
+            else:
+                if data in (b"II\x2a\x00", b"MM\x00\x2a", b"II\x2b\x00", b"MM\x00\x2b"):
+                    yield child
```

Comment:
Although I know it is not best practice to catch an error and then do nothing with it I do think its appropriate in this instance. We want to produce a list of tiffs here so if we cannot read it I think it is appropriate to just not include it in the list.
@manthey @brianhelba